code-challenges.test.js

```diff
@@ -15,35 +15,125 @@
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+
+    return array
+
+}
+
+
+
+
+// // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+// const nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// const nums2 = [109, 5, 9, 67, 8, 24]
+// // Expected output: [5, 109]
+
+describe('minMax', () => {
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order', () => {
+        expect(minMax(nums1)).toEqual([-8, 90]);
+        expect(minMax(nums2)).toEqual([5, 109]);
+    });
+  });
+
+
+// // b) Create the function that makes the test pass.
 
+// // create a function called "minMax"
+// // create param called "array"
+// // create a variable called "max" and "min"
+// // use reduce() method to obtain only 1 value
+// // create a function and use the Math.max/Math.min methods to obtain the max number and min number
+// // return the min and max number
 
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+const minMax = (array) => {
+    
+    var max = array.reduce(function(a, b) {
+        return Math.max(a, b) 
+    })
 
-// a) Create a test with an expect statement using the variables provided.
+    var min = array.reduce(function(a, b) {
+        return Math.min(a, b) 
+    })
+
+    return [min, max] 
+
+}
+
+
+ 
+
+
+// // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+
+// // a) Create a test with an expect statement using the variables provided.
+
+// const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+// const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+// // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+
+describe('combine', () => {
+    const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+    const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+    it('takes in two arrays as arguments and returns one array with no duplicate values', () => {
+      expect(combine(testArray1, testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1]);
+    });
+  });
 
-const testArray1 = [3, 7, 10, 5, 4, 3, 3]
-const testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
 
 // b) Create the function that makes the test pass.
+
+const combine = (array1, array2) => {
+
+    let combined = array1.concat(array2)
+    let uniqueChars = [];
+
+combined.forEach((element) => {
+    if (!uniqueChars.includes(element)) {
+        uniqueChars.push(element);
+    }
+
+});
+    return uniqueChars
+}
```

Comment:
Very cool! Just missing the pseudo code for this one!

Reply:
All done!